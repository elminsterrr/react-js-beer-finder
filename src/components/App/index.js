import React from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import beerLogo from './beer.svg';
import './style.css';

import StartButton from '../StartButton';

const App = () => (
  <MuiThemeProvider>
    <div className="App">
      <div>
        <header className="App-header">
          <h1 className="App-title">The Beer Finder</h1>
          <p className="App-subtitle">powered by love for</p>
          <img src={beerLogo} className="App-beer-logo" alt="logo" />
        </header>
        <div className="App-start-button">
          <StartButton />
        </div>
      </div>
    </div>
  </MuiThemeProvider>
);

export default App;
