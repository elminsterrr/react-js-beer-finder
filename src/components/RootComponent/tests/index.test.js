import React from 'react';
import { shallow } from 'enzyme';
import App from '../';

describe('App', () => {
  const app = shallow(<App />);

  it('renders correctly', () => {
    expect(app).toMatchSnapshot();
  });

  it('renders `App-start` children', () => {
    expect(app.find('.App-start').children().length).toEqual(1);
  });
});
