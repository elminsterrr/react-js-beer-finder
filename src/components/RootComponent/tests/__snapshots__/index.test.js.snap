// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RootComponent renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RootComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <header
                className="RootComponent-header"
        >
                <h1
                        className="RootComponent-title"
                >
                        BeerFinder
                </h1>
                <p
                        className="RootComponent-subtitle"
                >
                        powered by love for
                </p>
                <img
                        alt="logo"
                        className="RootComponent-beer-logo"
                        src="beer.svg"
                />
        </header>
        <div
                className="RootComponent-start"
        >
                <BrowserRouter>
                        <Connect(BeerListingEntry) />
                </BrowserRouter>
        </div>
</div>,
      "className": "RootComponent",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="RootComponent-header"
>
            <h1
                        className="RootComponent-title"
            >
                        BeerFinder
            </h1>
            <p
                        className="RootComponent-subtitle"
            >
                        powered by love for
            </p>
            <img
                        alt="logo"
                        className="RootComponent-beer-logo"
                        src="beer.svg"
            />
</header>,
          <div
            className="RootComponent-start"
>
            <BrowserRouter>
                        <Connect(BeerListingEntry) />
            </BrowserRouter>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="RootComponent-title"
>
                BeerFinder
</h1>,
              <p
                className="RootComponent-subtitle"
>
                powered by love for
</p>,
              <img
                alt="logo"
                className="RootComponent-beer-logo"
                src="beer.svg"
/>,
            ],
            "className": "RootComponent-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "BeerFinder",
                "className": "RootComponent-title",
              },
              "ref": null,
              "rendered": "BeerFinder",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "powered by love for",
                "className": "RootComponent-subtitle",
              },
              "ref": null,
              "rendered": "powered by love for",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "logo",
                "className": "RootComponent-beer-logo",
                "src": "beer.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <BrowserRouter>
              <Connect(BeerListingEntry) />
</BrowserRouter>,
            "className": "RootComponent-start",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Connect(BeerListingEntry) />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <header
                    className="RootComponent-header"
          >
                    <h1
                              className="RootComponent-title"
                    >
                              BeerFinder
                    </h1>
                    <p
                              className="RootComponent-subtitle"
                    >
                              powered by love for
                    </p>
                    <img
                              alt="logo"
                              className="RootComponent-beer-logo"
                              src="beer.svg"
                    />
          </header>
          <div
                    className="RootComponent-start"
          >
                    <BrowserRouter>
                              <Connect(BeerListingEntry) />
                    </BrowserRouter>
          </div>
</div>,
        "className": "RootComponent",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <header
              className="RootComponent-header"
>
              <h1
                            className="RootComponent-title"
              >
                            BeerFinder
              </h1>
              <p
                            className="RootComponent-subtitle"
              >
                            powered by love for
              </p>
              <img
                            alt="logo"
                            className="RootComponent-beer-logo"
                            src="beer.svg"
              />
</header>,
            <div
              className="RootComponent-start"
>
              <BrowserRouter>
                            <Connect(BeerListingEntry) />
              </BrowserRouter>
</div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1
                  className="RootComponent-title"
>
                  BeerFinder
</h1>,
                <p
                  className="RootComponent-subtitle"
>
                  powered by love for
</p>,
                <img
                  alt="logo"
                  className="RootComponent-beer-logo"
                  src="beer.svg"
/>,
              ],
              "className": "RootComponent-header",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "BeerFinder",
                  "className": "RootComponent-title",
                },
                "ref": null,
                "rendered": "BeerFinder",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "powered by love for",
                  "className": "RootComponent-subtitle",
                },
                "ref": null,
                "rendered": "powered by love for",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "logo",
                  "className": "RootComponent-beer-logo",
                  "src": "beer.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
            ],
            "type": "header",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <BrowserRouter>
                <Connect(BeerListingEntry) />
</BrowserRouter>,
              "className": "RootComponent-start",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Connect(BeerListingEntry) />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
